FAQ.md

```diff
@@ -0,0 +1,120 @@
+# Frequently Asked Questions
+(or, asked at least once anyway 😃)
+
+Not satisfied by the answer? Don't see your question? Swing by our
+[on Telegram](https://t.me/formality_lang) and we'd be happy to help.
+
+## Table of Contents
+
+Questions are in no particular order, but are loosely grouped
+by category.
+
+- [Frequently Asked Questions](#frequently-asked-questions)
+  - [Table of Contents](#table-of-contents)
+  - [Formality Features](#formality-features)
+    - [Indexed Data Types, or "What's that funny '~'"?](#indexed-data-types-or-whats-that-funny-)
+    - [Does Formality compile to native datatypes by name or definition?](#does-formality-compile-to-native-datatypes-by-name-or-definition)
+    - [How do I implement interfaces/typeclasses/traits?](#how-do-i-implement-interfacestypeclassestraits)
+    - [Is Formality Consistent?](#is-formality-consistent)
+    - [Can I have axioms?](#can-i-have-axioms)
+    - [What are the different parts of Formality?](#what-are-the-different-parts-of-formality)
+    - [What's the difference between Formality-Core and EA-TT?](#whats-the-difference-between-formality-core-and-ea-tt)
+    - [Does Formality use the Optimal Calculus (Ultimate Calculus), or does it use INets?](#does-formality-use-the-optimal-calculus-ultimate-calculus-or-does-it-use-inets)
+  - [Formality Concepts](#formality-concepts)
+    - [What's the difference between Bool.true and truth?](#whats-the-difference-between-booltrue-and-truth)
+    - [What is Unit? What is Empty? What does this have to do with truth and falsity?](#what-is-unit-what-is-empty-what-does-this-have-to-do-with-truth-and-falsity)
+    - [If Formality doesn't have side-effects, how do I do IO?](#if-formality-doesnt-have-side-effects-how-do-i-do-io)
+  - [Moonad](#moonad)
+    - [Have you heard of Unison's hash-based imports?](#have-you-heard-of-unisons-hash-based-imports)
+
+## Formality Features
+### Indexed Data Types, or "What's that funny '~'"?
+The `~` operator allows you to index a datatype. For example, 
+consider the definition of Vector, a List with a statically
+known length:
+```
+T Vec <A: Type>                 ~ (len: Nat)
+| nil                           ~ (zero);
+| cons(n:Nat, x:A, xs:Vec(A,n)) ~ (succ(n));
+```
+
+For comparison, in Agda the definition would be:
+
+```agda
+data Vec (A : Set) : (len : Nat) -> Set where
+  nil : Vec A len
```

Comment:
```suggestion
  nil : Vec A zero
```

Right?

Reply:
Yeah, that makes sense. Not to pass the buck, but it was copied and pasted from chat. We could probably rework the whole answer into something smoother.